src/metadata/gas.rs

```diff
@@ -39,8 +39,11 @@ pub enum GasMetadataError {
     ApChangeError(#[from] ApChangeError),
     #[error(transparent)]
     CostError(#[from] CostError),
-    #[error("not enough gas to run")]
-    NotEnoughGas,
+    #[error("Not enough gas to run the operation. Required: {required_gas}, Available: {available_gas}.")]
+    NotEnoughGas {
+        required_gas: u128,
+        available_gas: u128,
+    },
```

Comment:
This may bloat the error type a bit. Consider packing them in a box.

Reply:
Fixed